ibm/resource_ibm_database.go

```diff
@@ -0,0 +1,1086 @@
+	if err != nil {
+		return err
+	}
+	rsCatRepo := rsCatClient.ResourceCatalog()
+
+	serviceOff, err := rsCatRepo.FindByName(serviceName, true)
+	if err != nil {
+		return fmt.Errorf("Error retrieving database service offering: %s", err)
+	}
+
+	servicePlan, err := rsCatRepo.GetServicePlanID(serviceOff[0], plan)
+	if err != nil {
+		return fmt.Errorf("Error retrieving plan: %s", err)
+	}
+	rsInst.ServicePlanID = servicePlan
+
+	deployments, err := rsCatRepo.ListDeployments(servicePlan)
+	if err != nil {
+		return fmt.Errorf("Error retrieving deployment for plan %s : %s", plan, err)
+	}
+	if len(deployments) == 0 {
+		return fmt.Errorf("No deployment found for service plan : %s", plan)
+	}
+	deployments, supportedLocations := filterDatabaseDeployments(deployments, location)
+
+	if len(deployments) == 0 {
+		locationList := make([]string, 0, len(supportedLocations))
+		for l := range supportedLocations {
+			locationList = append(locationList, l)
+		}
+		return fmt.Errorf("No deployment found for service plan %s at location %s.\nValid location(s) are: %q.", plan, location, locationList)
+	}
+
+	rsInst.TargetCrn = deployments[0].CatalogCRN
+
+	if rsGrpID, ok := d.GetOk("resource_group_id"); ok {
+		rsInst.ResourceGroupID = rsGrpID.(string)
+	} else {
+		rsMangClient, err := meta.(ClientSession).ResourceManagementAPI()
+		if err != nil {
+			return err
+		}
+		resourceGroupQuery := management.ResourceGroupQuery{
+			Default: true,
+		}
+		grpList, err := rsMangClient.ResourceGroup().List(&resourceGroupQuery)
+		if err != nil {
+			return err
+		}
+		rsInst.ResourceGroupID = grpList[0].ID
+	}
+
+	params := Params{}
+	if memory, ok := d.GetOk("members_memory_allocation_mb"); ok {
+		params.Memory = memory.(int)
+	}
+	if disk, ok := d.GetOk("members_disk_allocation_mb"); ok {
+		params.Disk = disk.(int)
+	}
+	if version, ok := d.GetOk("version"); ok {
+		params.Version = version.(string)
+	}
+	if keyProtect, ok := d.GetOk("key_protect_key"); ok {
+		params.KeyProtectKey = keyProtect.(string)
+	}
+	parameters, _ := json.Marshal(params)
+	var raw map[string]interface{}
+	json.Unmarshal(parameters, &raw)
+	//paramString := string(parameters[:])
+	rsInst.Parameters = raw
+
+	if _, ok := d.GetOk("tags"); ok {
+		rsInst.Tags = getServiceTags(d)
+	}
+
+	instance, err := rsConClient.ResourceServiceInstance().CreateInstance(rsInst)
+	if err != nil {
+		return fmt.Errorf("Error creating database instance: %s", err)
+	}
+
+	// Moved d.SetId(instance.ID) to after waiting for resource to finish creation. Otherwise Terraform initates depedent tasks too early.
+	// Original flow had SetId here as its required as input to waitForDatabaseInstanceCreate
+
+	_, err = waitForDatabaseInstanceCreate(d, meta, instance.ID)
+	if err != nil {
+		return fmt.Errorf(
+			"Error waiting for create database instance (%s) to complete: %s", d.Id(), err)
+	}
+
+	icdId := EscapeUrlParm(instance.ID)
+	adminPassword := d.Get("adminpassword").(string)
```

Comment:
we are reading d.Get("adminpassowrd)...in 450 & 456 line

Reply:
Removed extra d.Get("adminpassword")